bitcoin/src/lib.rs

```diff
@@ -11,21 +11,23 @@
 //!
 //! See README.md for detailed documentation about development and supported environments.
 //!
-//! ## Available feature flags
+//! # Cargo features
 //!
-//! * `std` - the usual dependency on `std` (default).
-//! * `secp-recovery` - enables calculating public key from a signature and message.
-//! * `base64` - (dependency), enables encoding of PSBTs and message signatures.
-//! * `rand` - (dependency), makes it more convenient to generate random values.
-//! * `serde` - (dependency), implements `serde`-based serialization and
-//!                 deserialization.
+//! * `base64` (dependency) - enables encoding of PSBTs and message signatures.
+//! * `bitcoinconsensus` (dependency) - enables validating scripts and transactions.
+//! * `bitcoinconsensus-std` - same as `bitcoinconsensus` but also enables `std` here and in the
+//!                            `bitcoinconsensus` crate. Also communicates it to the `bitcoin` crate
```

Comment:
The hack was removed in #1978 and wasn't put back since then. Even I've forgotten about it being removed.

Basically, the feature enables returning `bitcoinconsensus::Error` from `std::error::Error` impl on `BitcoinconsensusError`.

[Your comment](https://github.com/rust-bitcoin/rust-bitcoin/pull/1978#issuecomment-1667922365) suggests it's a replacement but it really (also) changed semantics. Not in a wrong way but there's a footgun that someone might implicitly rely on it and then get broken code if the feature is off **or if the `bitcoinconsensus` version changes**.

So we should open an issue here but it's quite an edge case and probably annoying to fix anyway.

Reply:
I personally think all this is way to involved for the crate level docs, that is why I just removed it all previously.

Reply:
The hack definitely is. But people still need to know what the effect of the option is: "This allows returning `bitcoinconsensus::Error` from `std::error::Error::source`, but note that the library version may change!"